module/zfs/dsl_deadlist.c

```diff
@@ -875,28 +875,31 @@ dsl_deadlist_merge(dsl_deadlist_t *dl, uint64_t obj, dmu_tx_t *tx)
 		return;
 	}
 
+	za = kmem_alloc(sizeof (*za), KM_SLEEP);
+	pza = kmem_alloc(sizeof (*pza), KM_SLEEP);
```

Comment:
It would be more efficient to do a single allocation like this:

```
za = kmem_alloc(sizeof (*za) * 2, KM_SLEEP);
pza = &za[1];
```

Then we would free the memory by calling `kmem_free(za, sizeof (*za) * 2);` at the end of the function.

That said, I do not feel strongly about this, so I am fine with merging this either way. This is purely a suggestion for a way to minimize the runtime overhead of using dynamic memory to reduce stack space usage that stood out to me.